pandas/core/internals.py

```diff
@@ -937,10 +918,10 @@ def putmask(self, mask, new, align=True, inplace=False, axis=0,
 
         new_values = self.values if inplace else self.values.copy()
 
-        if hasattr(new, 'reindex_axis'):
```

Comment:
some of these may not be hit, I think I prefer the more idiomatic

``new = getattr(new, 'values', new)``

Reply:
Sure.  I'll check to see whether any value-less cases get through while I'm at it.